src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    if (herbivore.hidden === false) {
+      herbivore.health -= 50;
+    }
+
+    const index = Animal.alive.findIndex((item) => item.health <= 0);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
[CODE PERFORMANCE] The current implementation uses `findIndex` and `splice` to remove dead animals from the `Animal.alive` array, which is not the most efficient approach. It's preferred to use the `filter` method to avoid nesting loops and improve performance. Replace the `findIndex` and `splice` with a single `filter` operation to remove all dead animals at once.